### PR TITLE
[ACR] bump api version. support zone redundancy

### PR DESCRIPTION
**Pull request by adewaleo**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to review?

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def default_api_version(self):
             'role_definitions': '2018-01-01-preview',
             'provider_operations_metadata': '2018-01-01-preview'
         }),
-        ResourceType.MGMT_CONTAINERREGISTRY: '2019-12-01-preview',
+        ResourceType.MGMT_CONTAINERREGISTRY: '2020-11-01-preview',
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:165]: pls help to make sure run full live test for regression. :)

**Review by yungezz — COMMENTED** [review]
pls not checkin binary file

**Comment by adewaleo** [comment]
@yungezz thanks for the heads up and review. This is a wip pr with the generated sdk. I will remove it when the sdk has been published.

I will also make sure to run a full live test locally to ensure no regressions with the new api version.

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 1 above)
Comment (adewaleo) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:165]: I have re-run tests, please can you review @yungezz

**Comment by adewaleo** [comment]
@yungezz @fengzhou-msft  please can your review? i am hoping to make next week's code complete. Moreover some acr tests are broken because networking command loading / parameter registration is having issues. I have latest dev.

This section in [network/_params.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/network/_params.py#L120-L124) is causing problems:

```
        # region ApplicationGateways
        with self.argument_context('network application-gateway') as c:
            c.argument('application_gateway_name', app_gateway_name_type, options_list=['--name', '-n'])
>           c.argument('sku', arg_group='Gateway', help='The name of the SKU.', arg_type=get_enum_type(ApplicationGatewaySkuName), default=ApplicationGatewaySkuName.standard_medium.value)
E           AttributeError: 'NoneType' object has no attribute 'standard_medium'

```


this happens when tests like the one [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_commands.py#L510-L524) try to create a vnet:

```
During handling of the above exception, another exception occurred:

self = <azure.cli.command_modules.acr.tests.latest.test_acr_commands.AcrCommandsTests testMethod=test_acr_with_private_endpoint>, resource_group = 'clitest.rg000001'

    @ResourceGroupPreparer(location='centraluseuap')
    def test_acr_with_private_endpoint(self, resource_group):
        self.kwargs.update({
            'registry_name': self.create_random_name('testreg', 20),
            'vnet_name': self.create_random_name('testvnet', 20),
            'subnet_name': self.create_random_name('testsubnet', 20),
            'endpoint_name': self.create_random_name('priv_endpoint', 25),
            'endpoint_conn_name': self.create_random_name('priv_endpointconn', 25),
            'second_endpoint_name': self.create_random_name('priv_endpoint', 25),
            'second_endpoint_conn_name': self.create_random_name('priv_endpointconn', 25),
            'description_msg': 'somedescription'
        })

        # create subnet with disabled endpoint network policies
>       self.cmd('network vnet create -g {rg} -n {vnet_name} --subnet-name {subnet_name}')

src\azure-cli\azure\cli\command_modules\acr\tests\latest\test_acr_commands.py:537:
```

**Comment by yonzhan** [comment]
@jsntcy for awareness.

**Comment by jsntcy** [comment]
> @jsntcy Yu Chen FTE for awareness.

As we're migrating network to track2 SDK, hopefully the migration can be done by the end of this week and will check if this issue still exists after migration.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
LGTM

**Comment by adewaleo** [comment]
@yungezz please can you reapprove and merge when ready. I am no longer adding the `--bypass` option and tests and styliing should be good too.

**Review by fengzhou-msft — APPROVED** [review]

**Review by yungezz — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

**Pull request merged**